scripts/irods/test/test_misc.py

```diff
@@ -512,3 +508,31 @@ def test_configuration_validation_reports_fqdn_size_violations__issue_8287(self)
                 self.assertIn('"valid": false', res.stderr)
                 self.assertIn('"instanceLocation": "/host"', res.stderr)
                 self.assertIn('"error": "Number of characters must be at most 253"', res.stderr)
+
+    @unittest.skipIf(test.settings.TOPOLOGY_FROM_RESOURCE_SERVER, 'Test must be run from the Catalog Service Provider')
+    def test_server_returns_SYS_EXCEED_CONNECT_CNT_on_infinite_server_redirect_loop__issue_8529(self):
+        # This test requires that the Catalog Service Provider be identifiable by two names. This
+        # requirement is satisfied by the iRODS Testing Environment because setup_irods.py produces a
+        # server_config.json which initializes the "host" property to the container id while also allowing
+        # the same container to be identified (internally) as "irods-catalog-provider".
+
+        # Force the test to fail if it is not run from the provider.
+        hostname_alias = 'irods-catalog-provider'
+        with session.make_session_for_existing_admin(hostname_alias) as admin:
+            admin.assert_icommand(['imiscsvrinfo'], 'STDOUT', ['RCAT_ENABLED'])
```

Comment:
Does this rather (or, additionally) test that `irods-catalog-provider` is a valid hostname alias for the host running the test? We have the `skipIf` above to ensure that this is running on a catalog provider

Reply:
> Does this rather (or, additionally) test that `irods-catalog-provider` is a valid hostname alias for the host running the test?

That's correct.

Reply:
Okay. Maybe a comment with that info would be helpful here?

Reply:
Will include more words.

Reply:
Done.